libdrgn/program.c

```diff
@@ -164,14 +165,61 @@ drgn_program_check_initialized(struct drgn_program *prog)
 	return NULL;
 }
 
+#ifdef LIBKDUMPFILE
```

Comment:
Can you move this function to `kdump_reader.c`, again so we don't need the ifdef? Then we can also make a bunch of the kdump helpers static and will only need an ifdef in `kdump_reader.h` for this function.